Sources/PackageCollections/PackageCollections.swift

```diff
@@ -0,0 +1,296 @@
+        in profile: PackageCollectionsModel.Profile? = nil,
+        callback: @escaping (Result<PackageCollectionsModel.TargetListResult, Error>) -> Void
+    ) {
+        self.listCollections(identifiers: collections, in: profile) { result in
+            switch result {
+            case .failure(let error):
+                callback(.failure(error))
+            case .success(let collections):
+                let targets = self.targetListResultFromCollections(collections)
+                callback(.success(targets))
+            }
+        }
+    }
+
+    func findTargets(
+        _ query: String,
+        searchType: PackageCollectionsModel.TargetSearchType? = nil,
+        collections: Set<PackageCollectionsModel.CollectionIdentifier>? = nil,
+        profile: PackageCollectionsModel.Profile? = nil,
+        callback: @escaping (Result<PackageCollectionsModel.TargetSearchResult, Error>) -> Void
+    ) {
+        let profile = profile ?? .default
+        let searchType = searchType ?? .exactMatch
+
+        self.storage.collectionsProfiles.listSources(in: profile) { result in
+            switch result {
+            case .failure(let error):
+                callback(.failure(error))
+            case .success(let sources):
+                let identifiers = sources.map { .init(from: $0) }.filter { collections?.contains($0) ?? true }
+                if identifiers.isEmpty {
+                    return callback(.success(.init(items: [])))
+                }
+                self.storage.collections.searchTargets(identifiers: identifiers, query: query, type: searchType, callback: callback)
+            }
+        }
+    }
+
+    // FIXME:
+    func getPackageMetadata(_ reference: PackageReference,
+                            profile: PackageCollectionsModel.Profile? = nil,
+                            callback: @escaping (Result<PackageCollectionsModel.PackageMetadata, Error>) -> Void) {
+        fatalError("not implemented")
+    }
+
+    private func refreshPackageGroupFromSource(source: PackageCollectionsModel.CollectionSource,
+                                               order _: Int? = nil,
+                                               profile _: PackageCollectionsModel.Profile? = nil,
+                                               callback: @escaping (Result<PackageCollectionsModel.Collection, Error>) -> Void) {
+        if let errors = source.validate() {
+            return callback(.failure(MultipleErrors(errors)))
+        }
+        guard let provider = self.providers[source.type] else {
+            return callback(.failure(UnknownProvider(source.type)))
+        }
+        provider.get(source) { result in
+            switch result {
+            case .failure(let error):
+                callback(.failure(error))
+            case .success(let collection):
+                self.storage.collections.put(collection: collection, callback: callback)
```

Comment:
Let's add a code comment to explain we save package collection to local cache/storage to reduce network calls